src/scripts/main.js

```diff
@@ -1,3 +1,230 @@
 'use strict';
 
-// write your code here
+const button = document.querySelector('.button');
+const message = document.querySelector('.message-start');
+const lose = document.querySelector('.message-lose');
+const win = document.querySelector('.message-win');
+const rows = document.querySelectorAll('tr');
+const allCells = document.querySelectorAll('td');
+let buttonClicked = false;
+
+button.addEventListener('click', () => {
+  buttonClicked = true;
+  message.classList.add('hidden');
+
+  if (!buttonClicked) {
+    newCell();
+    newCell();
+  } else {
+    reset();
+  }
+});
+
+function isGameOver() {
+  let gameOver = true;
+
+  for (let i = 0; i < 4; i++) {
+    for (let j = 0; j < 4; j++) {
+      if (!rows[i].children[j].textContent) {
+        gameOver = false;
+        break;
+      }
+    }
+  }
+
+  if (gameOver === true) {
+    for (let i = 0; i < 4; i++) {
+      for (let j = 0; j < 3; j++) {
+        const checkCell = rows[i].children[j].textContent;
+        const nextCell = rows[i].children[j + 1].textContent;
+
+        if (checkCell === nextCell) {
+          gameOver = false;
+          break;
+        }
+      }
+    }
+  }
+
+  if (gameOver === true) {
```

Comment:
You defined three of the same conditional checks `if (gameOver === true)`, maybe we should use the all code in just one if if they required same condition ?

Reply:
at end of all conditional checks are code: 
`if (checkCell === nextCell) {
          gameOver = false;
          break;
        }`
if this condition will true, other conditional checks will not apply. Otherwise all three conditional checks will senseles apply.